core/src/main/java/bio/terra/pearl/core/service/participant/ProfileService.java

```diff
@@ -36,16 +39,38 @@ public Optional<Profile> loadWithMailingAddress(UUID profileId) {
     }
 
     @Transactional
-    public Profile updateWithMailingAddress(Profile profile) {
-        return dao.updateWithMailingAddress(profile);
-    }
+    public Profile updateWithMailingAddress(Profile profile, DataAuditInfo auditInfo) {
+        Profile updatedProfile = this.update(profile, auditInfo);
 
+        MailingAddress mailingAddress = profile.getMailingAddress();
+        if (mailingAddress != null) {
+            if (profile.getMailingAddressId() == null) {
+                // we've added a new mailing address
+                mailingAddress = mailingAddressDao.create(profile.getMailingAddress());
+                profile.setMailingAddressId(mailingAddress.getId());
+            } else {
+                mailingAddressDao.update(mailingAddress);
+            }
+        }
+
+        updatedProfile.setMailingAddress(mailingAddress);
+        return updatedProfile;
+    }
     @Transactional
-    public void delete(UUID profileId, Set<CascadeProperty> cascade) {
+    public void delete(UUID profileId, DataAuditInfo auditInfo) {
         Profile profile = dao.find(profileId).get();
-        dao.delete(profileId);
+        super.delete(profileId, auditInfo);
         if (profile.getMailingAddressId() != null) {
             mailingAddressDao.delete(profile.getMailingAddressId());
         }
     }
+
+    @Override
```

Comment:
The issue that this was fixing is that when the DataAuditedService ran `find` to get the old version of the model, `find` wouldn't include the MailingAddress. Because of that, every data audit made it look like the mailing address was being added anew, because it would go from null -> having a value instead of being updated or even not changed.